File: src/shared/schemas/protocol.py
from enum import Enum

"""
Define authorised terms to define things and states
"""


class DroneStatus(str, Enum):
    IDLE = "IDLE"  # Waiting at base
    MOVING = "MOVING"  # Flying towards a target
    DELIVERING = "DELIVERING"  # Dropping/loading
    CHARGING = "CHARGING"  # Charging
    MAINTENANCE = "MAINTENANCE"  # Out of service


class UrgencyLevel(str, Enum):
    STANDARD = "STANDARD"
    HIGH = "HIGH"  # Blood, Organs -> Absolute priority
    CRITICAL = "CRITICAL"  # Vital prognosis (Takes precedence over everything)


class ActionType(str, Enum):
    FLY_TO = "FLY_TO"  # Move
    PICKUP = "PICKUP"  # Load a package
    DROPOFF = "DROPOFF"  # Deliver a package
    CHARGE = "CHARGE"  # Recharge

File: shared/ts/src/schemas/protocol.ts
import { z } from "zod";

/**
 * Define authorised terms to define things and states
 * Mirrors: src/shared/schemas/protocol.py
 */

export const DroneStatusSchema = z.enum([
  "IDLE", // Waiting at base
  "MOVING", // Flying towards a target
  "DELIVERING", // Dropping/loading
  "CHARGING", // Charging
  "MAINTENANCE", // Out of service
]);

export type DroneStatus = z.infer<typeof DroneStatusSchema>;

export const UrgencyLevelSchema = z.enum([
  "STANDARD",
  "HIGH", // Blood, Organs -> Absolute priority
  "CRITICAL", // Vital prognosis (Takes precedence over everything)
]);

export type UrgencyLevel = z.infer<typeof UrgencyLevelSchema>;

export const ActionTypeSchema = z.enum([
  "FLY_TO", // Move
  "PICKUP", // Load a package
  "DROPOFF", // Deliver a package
  "CHARGE", // Recharge
]);

export type ActionType = z.infer<typeof ActionTypeSchema>;
